src/context/WalletProvider/MobileWallet/mobileMessageHandlers.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  createRevocableWallet,
+  RevocableWallet,
+} from 'context/WalletProvider/MobileWallet/RevocableWallet'
+import type {
+  MobileWalletInfo,
+  MobileWalletInfoWithMnemonic,
+} from 'context/WalletProvider/MobileWallet/types'
+
+import { mobileLogger } from './config'
+
+type Command =
+  | 'getWallet'
+  | 'deleteWallet'
+  | 'updateWallet'
+  | 'hasWallet'
+  | 'hasWallets'
+  | 'addWallet'
+  | 'listWallets'
+
+type Message =
+  | {
+      cmd: Command
+      key: string
+    }
+  | {
+      cmd: 'updateWallet'
+      key: string
+      label: string
+    }
+  | {
+      cmd: 'addWallet'
+      label: string
+      mnemonic: string
+    }
+  | {
+      cmd: 'listWallets' | 'hasWallets'
+    }
+
+export type MessageFromMobileApp = {
+  id: number
+  result: unknown
+}
+
+/**
+ * Create a Promise that sends a message and waits for the matching response
+ *
+ * This app uses `ReactNativeWebView.postMessage` to send a string to the mobile app
+ * The mobile app injects a `postMessage` call into the web app with a response
+ *
+ * Responses are matched to requests via an ID (Date.now())
+ *
+ * A 10 second timeout is included to avoid the Promise hanging
+ */
+const postMessage = async <T>(msg: Message): Promise<T> => {
+  return new Promise((resolve, reject) => {
+    const id = Date.now()
+    try {
+      const eventListener = (event: MessageEvent<MessageFromMobileApp>) => {
+        if (event.data?.id === id) {
+          window.removeEventListener('message', eventListener)
+          resolve(event.data.result as T)
+        }
+      }
+      // Make sure that the Promise doesn't hang forever
+      setTimeout(() => {
+        window.removeEventListener('message', eventListener)
+        reject(new Error('PostMessage timed out'))
+      }, 10000)
+
+      window.addEventListener('message', eventListener)
+
+      window.ReactNativeWebView?.postMessage(JSON.stringify({ ...msg, id }))
+    } catch (e) {
+      reject(e)
+    }
+  })
+}
+
+const moduleLogger = mobileLogger.child({ namespace: ['lib', 'mobileWallet'] })
+
+/**
+ * Get a list of available wallets stored in the mobile app
+ *
+ * The list does not include the mnemonic for the wallets
+ */
+export const listWallets = () => {
+  moduleLogger.trace({ fn: 'listWallets' }, 'List Wallets')
+  return postMessage<RevocableWallet[]>({ cmd: 'listWallets' })
+}
+
+/**
+ * Get a specific wallet by ID
+ *
+ * The ID is provided in the `listWallets` call
+ */
+export const getWallet = async (key: string) => {
+  moduleLogger.trace({ fn: 'getWallet', key }, 'Get Wallet')
+  const wallet = await postMessage<MobileWalletInfoWithMnemonic | null>({ cmd: 'getWallet', key })
+  return wallet ? createRevocableWallet(wallet) : null
+}
+
+/**
+ * Returns `true` if the mobile app has at least 1 saved wallet
+ */
+export const hasWallets = () => {
```

Comment:
can we please update comment and/or explicit return type, maybe `getWalletCount`?